Qt6: Could NOT find QtWaylandScanner

**Describe the bug**
A clear and concise description of what the bug is.
```
CMake Error at /usr/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find QtWaylandScanner (missing: QtWaylandScanner_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/ECM/find-modules/FindQtWaylandScanner.cmake:87 (find_package_handle_standard_args)
  src/platform/x11/systemclipboard/CMakeLists.txt:1 (find_package)
```
CMake expect to find `QtWaylandScanner_EXECUTABLE` in `/usr/bin/qtwaylandscanner` but it is in `/usr/lib64/qt6/qtwaylandscanner` instead for `qt6-wayland`.

**To Reproduce**
Steps to reproduce the behavior:
1. Build with `-DWITH_QT6:BOOL=ON` flag.

**Expected behavior**
A clear and concise description of what you expected to happen.
Building it from source code should work as it is working with Qt5.
I made a symbolic link of `/usr/lib64/qt6/qtwaylandscanner` to `/usr/bin/qtwaylandscanner`. To test if the build succeeds and it worked, but this is a problem because `/usr/bin/qtwaylandscanner` belongs to `qt5-wayland`

**Version, OS and Environment**
(Get details from `copyq version` command if possible.)
 - Application Version: 6.3.2
 - OS: Solus
 - Desktop environment: Budgie

--- a/src/platform/x11/systemclipboard/CMakeLists.txt
+++ b/src/platform/x11/systemclipboard/CMakeLists.txt
@@ -1,4 +1,3 @@
-find_package(QtWaylandScanner REQUIRED)
 include_directories(SYSTEM ${${copyq_qt}Gui_PRIVATE_INCLUDE_DIRS}) # for native interface to get wl_seat
 find_package(Wayland 1.15 COMPONENTS Client)
 find_package(${copyq_qt} ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS WaylandClient)
@@ -7,12 +6,21 @@ set(systemclipboard_SRCS
     waylandclipboard.cpp
 )
 
-ecm_add_qtwayland_client_protocol(systemclipboard_SRCS
-    PROTOCOL  wlr-data-control-unstable-v1.xml
-    BASENAME wlr-data-control-unstable-v1
-)
+if(WITH_QT6)
+    add_library(systemclipboard STATIC ${systemclipboard_SRCS})
+    qt_generate_wayland_protocol_client_sources(systemclipboard
+        FILES
+            "${CMAKE_CURRENT_SOURCE_DIR}/wlr-data-control-unstable-v1.xml"
+    )
+else()
+    find_package(QtWaylandScanner REQUIRED)
+    ecm_add_qtwayland_client_protocol(systemclipboard_SRCS
+        PROTOCOL wlr-data-control-unstable-v1.xml
+        BASENAME wlr-data-control-unstable-v1
+    )
+    add_library(systemclipboard STATIC ${systemclipboard_SRCS})
+endif()
 
-add_library(systemclipboard STATIC ${systemclipboard_SRCS})
 target_link_libraries(systemclipboard
                       ${copyq_qt}::Gui
                       ${copyq_qt}::WaylandClient

